.github/workflows/publish-to-npm.yaml

```diff
@@ -29,8 +29,28 @@ jobs:
             echo "has_changeset_files=false" >> "$GITHUB_OUTPUT"
           fi
 
-      - name: Setup Sui CLI and install dependencies
-        uses: ./.github/actions/install
+      - name: Get Sui Version
+        id: sui_version
+        shell: bash
+        run: |
+          SUI_VERSION=$(jq -r '.SUI_VERSION' version.json)
+          echo "sui_version=$SUI_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Setup Sui CLI
+        uses: ./.github/actions/install-sui-cli
+        with:
+          sui_version: ${{ steps.sui_version.outputs.sui_version }}
+
+      - name: Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+          cache: 'npm'
+          registry-url: 'https://registry.npmjs.org'
+  
+      - name: Install dependencies
+        shell: bash
+        run: npm ci
```

Comment:
create another setup-repo action that installs sui, and sets up npm to reduce duplication?